hazelcast/test/src/HazelcastTests3.cpp

```diff
@@ -990,6 +1008,9 @@ class BasicClientReplicatedMapNearCacheTest
 
     void create_no_near_cache_context()
     {
+        if (client_ != nullptr) {
+            client_->shutdown().get();
+        }
```

Comment:
These lines are OK but latest `client_` instance will be left as is. So `destructor` is needed.